tests/gtestmusig2.c

```diff
@@ -0,0 +1,435 @@
+        err = secp256k1_keypair_pub(ctx, &pk_list[i], &mcs_list[i].keypair);
+        if (err != 1) {
+            return err;
+        }
+        /* Store the batch commitments of the signer in batch_list */
+        l = 0; // the index of the signer's commitment list.
+        for (k = 0; k < NR_MESSAGES; k++) {
+            for (j = 0; j < V; j++, l++) {
+                ind = (nr_participants * V * k) + (j * nr_participants) + i; // the index for the list of collected batches.
+                err = secp256k1_keypair_pub(ctx, &batch_list[ind], mcs_list[i].comm_list[l]);
+                if (err != 1) {
+                    return err;
+                }
+            }
+        }
+    }
+    return 1;
+}
+
+static int aggregate_pk_batch(secp256k1_pubkey *pk_list, secp256k1_pubkey *batch_list, musig2_context_sig *mcs_list){
+    /**** Aggregate the public keys and batch commitments for each signer ****/
+    int i;
+    int cnt = 0;
+    for (i = 0; i < NR_SIGNERS; i++)
+        cnt += musig2_aggregate_pubkey(mcs_list[i].mc, pk_list, NR_SIGNERS);
+    if (cnt != NR_SIGNERS) {
+        return cnt;
+    }
+
+    cnt = 0;
+    for (i = 0; i < NR_SIGNERS; i++)
+        cnt += musig2_aggregate_R(&mcs_list[i], batch_list);
+    if (cnt != NR_SIGNERS) {
+        return cnt;
+    }
+    return 1;
+}
+
+static int sign_partial(musig2_context_sig *mcs_list, musig2_partial_signatures* mps, int nr_participants){
+    int i, err;
+    for (i = 0; i < nr_participants; i++) {
+        /* Generate the partial signatures */
+        err = musig2_sign(&mcs_list[i], &mps[i], MSG_1, MSG_1_LEN);
+        if (err != 1) {
+            return err;
+        }
+    }
+    return 1;
+}
+
+TEST (musig2, not_enough_signatures) {
+#define less_signers (NR_SIGNERS - 1)
+
+    int err;
+    secp256k1_context *ctx = secp256k1_context_create(SECP256K1_CONTEXT_VERIFY);
+    secp256k1_pubkey pk_list[NR_SIGNERS];    // Signers' public key list
+    secp256k1_pubkey batch_list[NR_SIGNERS * V * NR_MESSAGES];   // Stores the batches of signers
+    musig2_context_sig mcs_list[NR_SIGNERS]; // Array that holds NR_SIGNERS musig2_context_sig
+    musig2_partial_signatures mps[less_signers];
+    musig2_context mca;
+    unsigned char signature[SCH_SIG_BYTES];
+
+    // Init signers, store public keys, generate batch commitments for `NR_SIGNERS`.
+    err = init_musig2(ctx, pk_list, batch_list, mcs_list, NR_SIGNERS);
+    ASSERT_EQ(err, 1);
+
+    // Aggregate public keys and batch commitments.
+    err = aggregate_pk_batch(pk_list, batch_list, mcs_list);
+    ASSERT_EQ(err, 1);
+
+    // Generate partial signatures for `less_signers`.
+    err = sign_partial(mcs_list, mps, less_signers);
+    ASSERT_EQ(err, 1);
+
+    // Aggregate partial signatures for `NR_SIGNERS`.
+    // Aggregation should fail since there are fewer signatures than the registered signers.
+    err = musig2_aggregate_partial_sig(ctx, &mca, mps, pk_list, signature, NR_SIGNERS);
+    ASSERT_EQ(err, -2);
```

Comment:
The aggregation of signatures does not validate anymore if there are not enough signatures.